### PR TITLE
Breaking change: Moving item creation to the create_items step

### PR DESCRIPTION
**Pull request by FuzzyGamesOn**
This is something that's needed to be adjusted for a while. Currently, Manual creates its item pool in generate_basic, but it really should be created in create_items, since that's what it's there for. generate_basic is more there for post item pool things.

So this PR changes that. Some of the specific changes:
- There are now two hooks to replace before_generate_basic: before_create_items_starting and before_create_items_filler
   - That way, people can hook before whichever process they want during item creation
- The victory location and its event are now setup during create_regions instead of pre_fill, as another "what it's there for" change
- generate_basic now just handles place and don't place.

Updated all the hook templates and created new hooks for create_items, and took item pool out of generate basic hooks.

(As an aside, took out progressive items, since they were never implemented.)

### CODE REVIEW

File: src/__init__.py [hunk 1]
```diff
@@ -55,85 +54,71 @@ class ManualWorld(World):
...
-            #     shouldUseProgressive = (self.multiworld.progressive_items[self.player].value);
-
-            #     if shouldUseProgressive and name in self.progressive_item_table:
-            #         name = self.progressive_item_table[name]
+            if name == "__Victory__": continue
+            if name == filler_item_name: continue
 
             item = self.item_name_to_item[name]
-            item_count = 1
+            item_count = int(item.get("count") or 1)
```
Comment (silasary) [inline src/__init__.py:94]: This will increase `"count": 0` to 1.
You want to use `item.get("count", 1)`.

File: src/__init__.py [hunk 2]
```diff
@@ -313,8 +294,13 @@ def fill_slot_data(self):
 
         return slot_data
     
+    ###
+    # Non-standard AP world methods
+    ###
```
Comment (silasary) [inline src/__init__.py:299]: If I'm going to be technical, `generate_output` is a standard method

**Review by silasary — CHANGES_REQUESTED** [review]
Looks good, other than a couple minor tweaks

**Comment by FuzzyGamesOn** [comment]
> Looks good, other than a couple minor tweaks

Yep, got those sorted. Thanks for pointing those out! 😁 Forgot about generate_output down there, haha.

**Review by nicopop — APPROVED** [review]
looks good, ill need to modify stuff on my pr once this merge but this should work

**Review by silasary — APPROVED** [review]

**Pull request merged**